account_spread_cost_revenue/models/account_invoice_line.py

```diff
@@ -65,3 +66,39 @@ def spread_details(self):
             'target': 'new',
             'context': ctx,
         }
+
+    def create_auto_spread(self):
+        """ Create auto spread table for each invoice line, when needed """
+        for line in self:
+            if line.spread_check == 'linked':
+                continue
+            spread_type = (
+                line.invoice_type in ['out_invoice', 'out_refund'] and
+                'sale' or 'purchase')
+            spread_auto = self.env['account.spread.template.auto'].search(
+                [('template_id.auto_spread', '=', True),
+                 ('template_id.spread_type', '=', spread_type)])
+            matched = spread_auto.filtered(
+                lambda l:
+                (l.product_id == line.product_id and
+                 l.account_id == line.account_id)
+                or
+                (l.product_id == line.product_id and not l.account_id)
+                or
+                (l.account_id == line.account_id and not l.product_id)
+            )
+            if not matched:
+                continue
+            elif len(matched) > 1:
+                raise UserError(
+                    _('Too many auto spread options (%s) matched with the '
```

Comment:
If all matched relate to the same template this error should not be shown?

Reply:
> Since we are working strongly with invoices maybe it would also be good from a user perspective to add an additional smart button to spreads on invoices which contain them?

Not sure I understand this. You mean, add smart button in spread template, which open up all invoice line that match the auto spread criteria?

Reply:
No, more a generic comment. Add a smartbutton to account.invoice to show the spreads.

Reply:
> No, more a generic comment. Add a smartbutton to account.invoice to show the spreads.

But in account.invoice.line already have spread_id, you can link it, right?

Reply:
Any more comments?